### PR TITLE
Internal: Aligned release tools to the latest CKEditor 5 dev.

### PR DESCRIPTION
**Pull request by martnpaneq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Aligned release tools to the latest CKEditor 5 dev. Closes ckeditor/ckeditor5-package-generator#153.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### Supported scopes

* `generator` → https://www.npmjs.com/package/ckeditor5-package-generator
* `*` → https://www.npmjs.com/package/@ckeditor/ckeditor5-package-*

### CODE REVIEW

File: scripts/release/utils/parsearguments.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	options.npmTag = options[ 'npm-tag' ];
+	delete options[ 'npm-tag' ];
+
+	return options;
+};
+
+/**
+ * @typedef {Object} ReleaseOptions
+ *
+ * @property {Number} concurrency
```
Comment (psmyrek) [inline scripts/release/utils/parsearguments.js:44]: The `concurrency` argument is not supported in this package.

File: scripts/release/utils/parsearguments.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * @property {String} [npmTag='staging']
```
Comment (psmyrek) [inline scripts/release/utils/parsearguments.js:46]: The default tag is `latest`.

**Review by psmyrek — COMMENTED** [review]
Changes work fine 🎉 

Could you also update all `@ckeditor/ckeditor5-dev-*` versions to `^38.0.0` in `packages/ckeditor5-package-generator/package.json` and `packages/ckeditor5-package-tools/package.json`?

File: scripts/release/preparepackages.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		task: () => {
+			return releaseTools.cleanUpPackages( {
+				packagesDirectory: RELEASE_DIRECTORY,
+				packageJsonFieldsToRemove: [
+					// TODO: Preserve `scripts.postinstall`.
+					// See: https://github.com/ckeditor/ckeditor5/issues/14318.
+					'devDependencies',
+					'depcheckIgnore'
+				]
+			} );
```
Comment (psmyrek) [inline scripts/release/preparepackages.js:68]: Do we need the `postinstall` script in a package published on npm?

Comment (martnpaneq) [reply scripts/release/preparepackages.js:68]: Good catch with the comments.

I'm just not sure about publishing the `postinstall` hook. We are publishing it for example in react and vue integrations:
https://www.npmjs.com/package/@ckeditor/ckeditor5-react?activeTab=code
https://www.npmjs.com/package/@ckeditor/ckeditor5-vue?activeTab=code

So I'm adding it to preserve the convention.

**Review by psmyrek — APPROVED** [review]
LGTM 🎉

**Review by pomek — CHANGES_REQUESTED** [review]
I think, in the `preparepackages.js` script, we miss updating the version of dependencies between packages within the mono repository.

**Comment by martnpaneq** [comment]
> I think, in the `preparepackages.js` script, we miss updating the version of dependencies between packages within the mono repository.

@pomek Added cross package dependency update.

**Comment by pomek** [comment]
A callback for determing if a package belongs to the mono repository must be smarter, as the repo contains both: scoped and non-scoped packages. See:

* https://github.com/ckeditor/ckeditor5-package-generator/blob/72720d04cc84ee0c65b11dcd8482613ba957007f/packages/ckeditor5-package-generator/package.json#L2
* https://github.com/ckeditor/ckeditor5-package-generator/blob/72720d04cc84ee0c65b11dcd8482613ba957007f/packages/ckeditor5-package-tools/package.json#L2

**Review by pomek — APPROVED** [review]
LGTM.

**Pull request merged**